File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03998/p03998_num7_parsed.cpp
#include <iostream>
#include <queue>
using namespace std;

int main() {
    string S_A, S_B, S_C;
    cin >> S_A >> S_B >> S_C;

    queue<int> deck_A, deck_B, deck_C;

    for (int i = 0; i < S_A.size(); i++) {
        deck_A.push(S_A[i]);
    }

    for (int i = 0; i < S_B.size(); i++) {
        deck_B.push(S_B[i]);
    }

    for (int i = 0; i < S_C.size(); i++) {
        deck_C.push(S_C[i]);
    }

    char current_player = 'A';

    while (!deck_A.empty() && !deck_B.empty() && !deck_C.empty()) {
        if (current_player == 'A') {
            char card_top = deck_A.front();
            deck_A.pop();
            current_player = card_top;
        } else if (current_player == 'B') {
            char card_top = deck_B.front();
            deck_B.pop();
            current_player = card_top;
        } else if (current_player == 'C') {
            char card_top = deck_C.front();
            deck_C.pop();
            current_player = card_top;
        }
    }

    if (deck_A.empty()) {
        cout << "A" << endl;
    } else if (deck_B.empty()) {
        cout << "B" << endl;
    } else if (deck_C.empty()) {
        cout << "C" << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03998/p03998_num7_parsed.py
deck_a = input().strip()
deck_b = input().strip()
deck_c = input().strip()

current_player = 'A'

while True:
    if current_player == 'A':
        if len(deck_a) == 0:
            print('A')
            break
        else:
            current_card = deck_a[0]
            deck_a = deck_a[1:]
            current_player = 'A' if current_card == 'a' else 'B' if current_card == 'b' else 'C'
    elif current_player == 'B':
        if len(deck_b) == 0:
            print('B')
            break
        else:
            current_card = deck_b[0]
            deck_b = deck_b[1:]
            current_player = 'A' if current_card == 'a' else 'B' if current_card == 'b' else 'C'
    elif current_player == 'C':
        if len(deck_c) == 0:
            print('C')
            break
        else:
            current_card = deck_c[0]
            deck_c = deck_c[1:]
            current_player = 'A' if current_card == 'a' else 'B' if current_card == 'b' else 'C'